File: samples/simple/assets/scripts/cube.lua
local transform_ = GetTransform();

--@serializable
speed_ = 25.0

space_pressed_ = false

function IsZeroVector3(vec)
  return vec.x == 0.0 and vec.y == 0.0 and vec.z == 0.0;
end

function OnStart()
  print("Hello from " .. GetName())
end

function OnUpdate(ds)
  local direction = Vec3:new()

  if IsKeyPressed(KeyCode.W) then
    direction = direction + transform_:GetForward();
  end
  if IsKeyPressed(KeyCode.S) then
    direction = direction - transform_:GetForward();
  end

  if IsKeyPressed(KeyCode.D) then
    direction = direction + transform_:GetRight();
  end
  if IsKeyPressed(KeyCode.A) then
    direction = direction - transform_:GetRight();
  end

  if not space_pressed_ and IsKeyPressed(KeyCode.Space) then
    SceneManager.SetActive(1);
    space_pressed_ = true;
  end

  if not IsZeroVector3(direction) then
    transform_:Translate(
      Mathf.Normalize(direction) * speed_ * ds
    );
  end
end

File: samples/simple/assets/scripts/Cube.cs
using System;
using EveEngine;

namespace Simple
{
    public class Cube : Entity
    {
        private bool _spacePressed = false;

        public float speed = 10.0f;

        protected override void OnCreate()
        {
            for (int i = 0; i < 10; i++)
            {
                Debug.Log(String.Format("Hello, World! {0}", i));
            }
        }

        protected override void OnUpdate(float ds)
        {
            Vector3 direction = Vector3.Zero;

            if (Input.IsKeyPressed(KeyCode.W))
            {
                direction += Transform.GetForward();
            }
            if (Input.IsKeyPressed(KeyCode.S))
            {
                direction -= Transform.GetForward();
            }

            if (Input.IsKeyPressed(KeyCode.D))
            {
                direction += Transform.GetRight();
            }
            if (Input.IsKeyPressed(KeyCode.A))
            {
                direction -= Transform.GetRight();
            }

            if (!_spacePressed && Input.IsKeyPressed(KeyCode.Space))
            {
                SceneManager.SetActive(1);
                _spacePressed = true;
            }

            if (direction != Vector3.Zero)
            {
                Transform.Position += Mathf.Normalize(direction) * speed * ds;
            }
        }
    }
}
